### ISSUE TITLE
No warning given when explicit method signatures mismatch

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#180]
This gives a fatal error in PHP 7:

```php
abstract class A {
  abstract public function foo() : string;
}

class B extends A
{
  public function foo() {
    return "hello";
  }
}
```

But Psalm says nothing, treats it as if the `string` return type is equivalent to the same in a docblock.

Ditto with 
```php
interface A {
  public function foo() : string;
}

class B implements A
{
  public function foo() {
    return "hello";
  }
}
```

**Comment by muglug** [comment]
This is no longer an issue
